pep-0688.rst

```diff
@@ -0,0 +1,258 @@
+in a pure Python package. As a temporary workaround, a
+``typing_extensions.Buffer`` ABC will be provided on Python versions
+that do not have ``types.Buffer`` available. For the benefit of
+static type checkers, ``typing_extensions.Buffer`` can be used as
+a base class in stubs to mark types as supporting the buffer protocol.
+For runtime uses, the ``ABC.register`` API can be used to register
+buffer classes with ``typing_extensions.Buffer``. When
+``types.Buffer`` is available, ``typing_extensions`` should simply
+re-export it.
+
+
+No special meaning for ``bytes``
+--------------------------------
+
+The special case stating that ``bytes`` may be used as a shorthand
+for other ``ByteString`` types will be removed from the ``typing``
+documentation.
+With ``types.Buffer`` available as an alternative, there is no good
+reason to allow ``bytes`` as a shorthand.
+We suggest that type checkers that implement this behavior should deprecate and
+eventually remove it.
+
+
+Backwards Compatibility
+=======================
+
+As the runtime changes in this PEP only add a new class, there are
+no backwards compatibility concerns.
+
+However, the recommendation to remove the special behavior for
+``bytes`` in type checkers does have backwards compatibility
+impact on users. An `experiment <https://github.com/python/mypy/pull/12661>`__
+with mypy shows that several major open source projects type
+checked with mypy will see new errors if the ``bytes`` promotion
+is removed. However, many of these errors can be fixed by improving
+the stubs in typeshed, as already done for the
+`builtins <https://github.com/python/typeshed/pull/7631>`__,
+`binascii <https://github.com/python/typeshed/pull/7677>`__,
+`pickle <https://github.com/python/typeshed/pull/7678>`__ modules.
+Overall, the change improves overall type safety,
+so we believe the migration cost is worth it.
+
+
+Security Implications
+=====================
+
+None.
+
+
+How to Teach This
+=================
+
+We will add notes pointing to ``types.Buffer`` to appropriate places in the
+documentation, such as `typing.readthedocs.io <https://typing.readthedocs.io/en/latest/>`__
+and the `mypy cheat sheet <https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html>`__.
+Type checkers may provide additional pointers in their error messages. For example,
+when they encounter a place where a buffer object is passed to a function that
+is annotated to only accept ``bytes``, the error message could include a note suggesting
+to use ``types.Buffer`` instead.
+
+
+Reference Implementation
+========================
+
+An implementation of ``types.Buffer`` is
+`available <https://github.com/python/cpython/compare/main...JelleZijlstra:typesbuffer?expand=1>`__
+in the author's fork.
+
+
+Rejected Ideas
+==============
+
+Buffer ABC
+----------
+
+An `earlier proposal <https://github.com/python/cpython/issues/71688>`__ suggested
+adding a ``collections.abc.Buffer`` ABC to represent buffer objects. This idea
```

Comment:
```suggestion
adding a ``collections.abc.Buffer`` `abstract base class <https://docs.python.org/3/glossary.html#term-abstract-base-class>`__
to represent buffer objects. This idea
```

As before, might be good actually mention once what "ABC" means and link to it, in case any readers are unfamiliar